src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -1594,14 +1610,28 @@ protected void CheckModifiersForBody(Location location, DiagnosticBag diagnostic
 
         internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree localTree)
         {
-            // Method without body doesn't declare locals.
-            Debug.Assert(this.BodySyntax != null);
-            Debug.Assert(this.BodySyntax.SyntaxTree == localTree);
+            Debug.Assert(this.SyntaxNode.SyntaxTree == localTree);
+
+            (BlockSyntax, ArrowExpressionClauseSyntax) bodies = Bodies;
```

Comment:
Perhaps deconstruct: `(BlockSyntax block, ArrowExpressionClauseSyntax arrow) = Bodies;` to avoid using `Item1` or `Item2`.